From 154c536c711de1c09edbd880820da01c5b457a6c Mon Sep 17 00:00:00 2001
From: Thomas Lamprecht <t.lamprecht@proxmox.com>
Date: Thu, 30 Jan 2020 12:48:23 +0100
Subject: [PATCH 06/12] networking.service: fix dependencies and ordering

IMPORTANT: DO NOT REMOVE IF UNSURE!

ONLY IF IFUPDOWN2 FIXED THEIR SERVICE MESS.

THEY CLOSED TWO PULL REQUESTS ALREADY WITHOUT REASON.

SO WE JUST HAVE TO KEEP THIS DOWNSTREAM PATCH!

BIG WARNING HERE IS DONE BECAUSE IT WAS ALREADY REMOVED BY MISTAKE
ONCE!!

Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 debian/ifupdown2-pre.service        | 13 +++++++++++++
 debian/ifupdown2.networking.service |  8 +++++---
 debian/rules                        |  1 +
 3 files changed, 19 insertions(+), 3 deletions(-)
 create mode 100644 debian/ifupdown2-pre.service

diff --git a/debian/ifupdown2-pre.service b/debian/ifupdown2-pre.service
new file mode 100644
index 0000000..b129a62
--- /dev/null
+++ b/debian/ifupdown2-pre.service
@@ -0,0 +1,13 @@
+[Unit]
+Description=Helper to synchronize boot up for ifupdown
+DefaultDependencies=no
+Wants=systemd-udevd.service
+After=systemd-udev-trigger.service
+Before=network.target
+
+[Service]
+Type=oneshot
+TimeoutSec=180
+RemainAfterExit=yes
+EnvironmentFile=-/etc/default/networking
+ExecStart=/bin/udevadm settle
diff --git a/debian/ifupdown2.networking.service b/debian/ifupdown2.networking.service
index cdb3065..ee5f17a 100644
--- a/debian/ifupdown2.networking.service
+++ b/debian/ifupdown2.networking.service
@@ -5,8 +5,9 @@ DefaultDependencies=no
 After=local-fs.target network-pre.target
 Before=shutdown.target network.target network-online.target
 Conflicts=shutdown.target
-Wants=systemd-udev-settle.service
-After=systemd-udev-settle.service
+Requires=ifupdown2-pre.service
+Wants=network.target
+After=local-fs.target network-pre.target systemd-sysctl.service systemd-modules-load.service ifupdown2-pre.service
 
 [Service]
 Type=oneshot
@@ -19,4 +20,5 @@ ExecStop=/usr/share/ifupdown2/sbin/start-networking stop
 ExecReload=/usr/share/ifupdown2/sbin/start-networking reload
 
 [Install]
-WantedBy=basic.target network.target shutdown.target
+WantedBy=multi-user.target network-online.target
+
diff --git a/debian/rules b/debian/rules
index ea6f434..fcb304f 100755
--- a/debian/rules
+++ b/debian/rules
@@ -15,6 +15,7 @@ override_dh_install:
 	dh_install
 	mkdir -p debian/ifupdown2/lib/systemd/system/
 	install --mode=644 debian/ifup@.service debian/ifupdown2/lib/systemd/system/
+	install --mode=644 debian/ifupdown2-pre.service debian/ifupdown2/lib/systemd/system/
 
 
 override_dh_systemd_start:
-- 
2.30.2

